.gitlab-ci.yml

```diff
@@ -0,0 +1,50 @@
+# Ruby GitLab configuration file
+#
+# Check [GITLAB SITE] for more details
+#
+default:
+  image: ruby:2.7.3
+
+cache:
+  files:
+    - Gemfile.lock
+  paths:
+    - vendor/ruby
+
+before_script:
+  - bundle config set path 'vendor'
+
+stages:
+  - build
+  - test
+  - deploy
+
+build:
+  stage: build
+  image: ruby:2.7.3
+  script:
+    - bundle install --deployment --jobs=4 --retry=3 --without deploy development doc production --path vendor/ruby
+    - bundle exec rake login:deploy_json
+
+test:
+  script:
+    - make test
+  artifacts:
+    paths:
+      - /tmp/test-results
+
+deploy:
+  stage: deploy
+  image: ruby:2.7.3
+  script:
+    -  curl -v -L -o cf-cli_amd64.deb 'https://cli.run.pivotal.io/stable?release=debian64&source=github'
+    - sudo dpkg -i cf-cli_amd64.deb
+    - cf -v
+    - bundle install --deployment --jobs=4 --retry=3 --without deploy development doc production --path vendor/ruby
+    - bundle exec rake login:deploy_json
+    - cf login -a https://api.fr.cloud.gov -u "4fbf18ec-1497-433f-814f-8c175041a6f1" -p $CF_PASS -o "gsa-login-prototyping" -s "$SPACE"
+    - cf push $SPACE-identity-saml-sinatra
```

Comment:
might want to see if we can feature flag this, so that we don't try to deploy from both CircleCI and Gitlab at the same time?